pkg/services/ngalert/state/state.go

```diff
@@ -87,13 +89,26 @@ func (a *State) resultAlerting(alertRule *ngModels.AlertRule, result eval.Result
 
 func (a *State) resultError(alertRule *ngModels.AlertRule, result eval.Result) {
 	a.Error = result.Error
+
 	if a.StartsAt.IsZero() {
 		a.StartsAt = result.EvaluatedAt
 	}
 	a.setEndsAt(alertRule, result)
 
 	if alertRule.ExecErrState == ngModels.AlertingErrState {
 		a.State = eval.Alerting
+	} else if alertRule.ExecErrState == ngModels.ErrorErrState {
+		a.State = eval.Error
+
+		// If the evaluation failed because a query returned an error then
+		// update the state with the Ref ID as a label and the error message
+		// as an annotation so other code can use this metadata to add context
+		// to alerts
+		var queryError expr.QueryError
+		if errors.As(a.Error, &queryError) {
+			a.Labels["RefId"] = queryError.RefID
```

Comment:
Here the key is `"RefId"`, but in `pkg/services/ngalert/schedule/compat.go` it's `"RefID"`.

Reply:
As far as I can tell we use `RefId` in the API so I thought it makes sense to also use it as the label name. `RefID` is just used in the Go code I think?